platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/LocationFragmentActivity.kt

```diff
@@ -116,7 +116,9 @@ class LocationFragmentActivity : AppCompatActivity() {
     }
 
     override fun onSuccess(result: LocationEngineResult?) {
+      if (!mapView.isDestroyed) {
```

Comment:
I think that instead of checking whether the map has been destroyed, the whole callback should hold a weak reference to the fragment which will prevent the leak in general.

Reply:
Thanks for the comments! The reason behind this check is that I got the following error during the sanity tests: 

`com.mapbox.mapboxsdk.MapStrictModeException: Map detected an error that would fail silently otherwise: You're calling `cancelTransitions` after the `MapView` was destroyed, were you invoking it after `onDestroy()`?`

It seems even after the mapView has been destroyed, the mapboxMap is still available for user to use, shall we set mapboxMap to null during the mapView `onDestroy` process?